runtime/src/main/kotlin/konan/worker/Atomics.kt

```diff
@@ -123,6 +124,7 @@ external private fun checkIfFrozen(ref: Any?)
  * Otherwise memory leak could happen.
  */
 @Frozen
+@NoReorderFields
```

Comment:
Also add this on WeakReferenceCounter in WeakPrivate.kt, it follows the same pattern.

Reply:
done